src/librustc_typeck/check/op.rs

```diff
@@ -223,13 +223,18 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                         };
 
                         if let Some(missing_trait) = missing_trait {
-                            if missing_trait == "std::ops::Add" {
-                                self.check_str_addition(expr, lhs_expr, lhs_ty,
-                                                        rhs_expr, rhs_ty_var, &mut err);
+                            // The logic here is saying if the trait was an add
+                            // and it was an addition with &str then skip the
+                            // span note below it. Otherwise just emit the span
+                            // note
+                            if missing_trait == "std::ops::Add" &&
```

Comment:
You're right. I got the logic wrong here. I tested it just to make sure so this will need to get changed